hadoop-yarn-project/pom.xml

```diff
@@ -90,6 +91,56 @@
       <artifactId>hadoop-yarn-applications-catalog-webapp</artifactId>
       <type>war</type>
     </dependency>
+    <dependency>
+      <groupId>org.apache.hadoop</groupId>
+      <artifactId>hadoop-yarn-applications-distributedshell</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>org.apache.hadoop</groupId>
+      <artifactId>hadoop-yarn-applications-unmanaged-am-launcher</artifactId>
+      <version>${project.version}</version>
+    </dependency>
+    <dependency>
+      <groupId>org.apache.hadoop</groupId>
+      <artifactId>hadoop-yarn-server-tests</artifactId>
+      <version>${project.version}</version>
+    </dependency>
+    <dependency>
+      <groupId>org.apache.hadoop</groupId>
+      <artifactId>hadoop-yarn-server-timelineservice-hbase-client</artifactId>
+      <version>${project.version}</version>
+      <exclusions>
```

Comment:
this worries me. why is this exclusion needed?

Reply:
cd hadoop-yarn-project

mvn clean -T2C -Pnative -Pdist -Dtar  -Psrc -Pyarn-ui -Dzookeeper.version=3.7.2 -Dhbase.profile=2.0 -DskipTests -DskipITs install 

The purpose of adding exclusions here is to resolve version conflicts during compilation, as shown in the diagram. I did not encounter this conflict when testing with Hadoop 3.3.6; it only appears in the trunk branch.
![image](https://github.com/apache/hadoop/assets/18082602/4e558bee-6218-4036-baee-8cf26109c550)